### DISCUSSION TITLE
Controller not updating value when using BaseUI Select

### DISCUSSION THREAD

**Discussion by asciant** [react-hook-form/react-hook-form#4508] [Q&A] [answered]
I have an issue where I am able to get a BaseUI Select functioning as expected when wrapped in a Controller from RHF, although when I submit the form the value form the input remains at the default value (it doesn't match what is selected in the Select component).

I tried to update the example from RHFI (https://github.com/react-hook-form/input/issues/16) and update it to the new Controller and BaseUI Select functionality. The CSB is here to view -> https://codesandbox.io/s/select-forked-iponp?file=/example.js

As you can see, the Select seems to function as expected. Although clicking submit results in "Beige" being returned as the value for the `color` input, which doesn't match what is selected by the Select component.

I'm guessing that I have missed something, and probably that the Controller has no idea what the select State variable is set to, thus it can't update the form value. I figured I could use a `setValue` call from the `useForm` hook, although I thought the Controller component did this with some internal magic. 

Any assistance appreciated.

**Comment by bluebill1049** [comment, marked as answer]
you will need to use `onChange` from controller to update value inside the hoook.

**Reply by asciant** [reply]
Thanks for the reply Bill.

I was able to use your information to get the CSB working.
